File: src/main/java/no/nav/domain/pensjon/regler/grunnlag/Uforehistorikk.java
package no.nav.domain.pensjon.regler.grunnlag;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.Collections;
import java.util.Date;
import java.util.List;

public class Uforehistorikk implements Serializable {

    

    /**
     * Liste av uføreperioder.
     */
    private List<Uforeperiode> uforeperiodeListe;

    /**
     * Uføregraden pensjonen er blitt fryst fra.
     */
    private int garantigrad;

    /**
     * Yrkesskadegraden pensjonen er blitt fryst fra.
     */
    private int garantigradYrke;

    /**
     * Dato for sist innmeldt i Folketrygden- for fremtidig trygdetid.
     * Lagt inn ifm PENPORT-2222
     */
    private Date sistMedlITrygden;

    /**
     * Ved eksport og ung uføre vil denne listen inneholde uførehistorikk med ung uføre.
     * Intern PREG variabel
     */
    private List<Uforeperiode> PREG_originalUforeperiodeListe;



}

File: src/main/java/no/nav/pensjon/regler/domain/grunnlag/Uforehistorikk.kt
package no.nav.pensjon.regler.domain.grunnlag

import java.io.Serializable
import java.util.*

class Uforehistorikk : Serializable {
    /**
     * Liste av Uføreperioder.
     */
    var uforeperiodeListe: List<Uforeperiode> = mutableListOf()

    /**
     * Uføregraden pensjonen er blitt fryst fra.
     */
    var garantigrad = 0

    /**
     * Yrkesskadegraden pensjonen er blitt fryst fra.
     */
    var garantigradYrke = 0

    /**
     * Dato for sist innmeldt i Folketrygden- for fremtidig trygdetid.
     * Lagt inn ifm PENPORT-2222
     */
    var sistMedlITrygden: Date? = null

}